caffe2/core/dispatch/LayoutId.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "caffe2/utils/IdWrapper.h"
+
+namespace c10 {
+
+class LayoutId final : public c10::guts::IdWrapper<LayoutId, uint8_t> {
+public:
+    constexpr explicit LayoutId(underlying_type id): IdWrapper(id) {}
+
+    // Don't use this default constructor!
+    // Unfortunately, a default constructor needs to be defined because of https://reviews.llvm.org/D41223
+    constexpr LayoutId(): IdWrapper(0) {}
```

Comment:
If we mark this default constructor as deprecated (there's a new ATen macro for doing this portably), will that help catch use-sites of it?

Reply:
Possibly. Not changing it here because I'm already working on a follow-up PR that removes LayoutId.